distribution/bwc-zip/build.gradle

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.elasticsearch.gradle.LoggedExec
+
+/**
+ * This is a dummy project which does a local worktree checkout of the previous
+ * major version's stable branch, and builds a snapshot. This allows backcompat
+ * tests in the next major version to test against the next unreleased minor
+ * version, without relying on snapshots.
+ */
+
+apply plugin: 'distribution'
+
+// TODO: generate this, by introspecting Version.java for last previous minor
+String BWC_VERSION = "5.3.0-SNAPSHOT"
+
+// cleanup any previous worktree refs which may have been removed by clean task
+task pruneWorktree(type: LoggedExec) {
+  commandLine = ['git', 'worktree', 'prune']
+}
+
+String worktreeDir = "${buildDir}/bwc/checkout-5.x"
+task createWorktree(type: LoggedExec) {
+  dependsOn pruneWorktree
+  onlyIf { new File(worktreeDir).exists() == false }
+  commandLine = ['git', 'worktree', 'add', '--detach', worktreeDir, '5.x']
+}
+
+task updateWorktree(type: LoggedExec) {
+  dependsOn createWorktree
+  commandLine = ['git', 'pull', 'origin', '5.x']
```

Comment:
I know we have had trouble in the past with people that do not name their remote to this repository `origin`. I think that we can avoid it here if we parse the output `git remote -v` and find the entry that corresponds to `elastic/elasticsearch` to get the right remote name?